src/xmipp/libraries/tomo/tomo_ctf_wiener2d_correction.cpp

```diff
@@ -0,0 +1,203 @@
+/***************************************************************************
+ *
+ * Authors:     Jose Luis Vilas (jlvilas@cnb.csic.es)
+ *
+ * Spanish Research Council for Biotechnology, Madrid, Spain
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#include "tomo_ctf_wiener2d_correction.h"
+#include <sys/stat.h>
+#include "data/ctf.h"
+#include <core/metadata_extension.h>
+#include "reconstruction/ctf_correct_wiener2d.h"
+#include "core/transformations.h"
+#include <limits>
+#include <type_traits>
+
+
+void ProgCTFWiener2DCorrection::defineParams()
+{
+	addUsageLine("This method performs a CTF correction of a tilt series");
+	addParamsLine("   -i <input_file>                    : Metadata with the tilt series");
+	addParamsLine("   -o <output_volume>          	 	 : Output volume.");
+	addParamsLine("   --sampling <sampling=1>            : Pixel size (Angstrom)");
+	addParamsLine("   [--wiener_constant <wc=0.1>        : Wiener constant");
+	addParamsLine("   [--threads <nthreads=1>]           : (Optional) Number of threads to be used");
```

Comment:
I see the threads parameter but no parallelization in the code. This code could be easily parallelized with MPI, but in that case this parameter is not necessary.

Reply:
Absolutely, I wanted to parallerize the code and then I left it for the future.... Sorry